PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:6 + add:0, change matches:
-    _inherit = ['account.move.line']
-    _inherit = ['purchase.order']
-    _inherit = ['stock.move']
-    _inherit = ['stock.picking']
-    _inherit = ['stock.rule']
-    _inherit = ['stock.valuation.layer']

Total Changes: 12

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/mrp_subcontracting_purchase/models/account_move_line.py
+++ b/addons/mrp_subcontracting_purchase/models/account_move_line.py
@@ -5,7 +5,7 @@ from odoo.tools import float_is_zero
 
 
 class AccountMoveLine(models.Model):
-    _inherit = ['account.move.line']
+    _inherit = 'account.move.line'
 
     def _get_price_unit_val_dif_and_relevant_qty(self):
         price_unit_val_dif, relevant_qty = super()._get_price_unit_val_dif_and_relevant_qty()

--- a/addons/mrp_subcontracting_purchase/models/purchase_order.py
+++ b/addons/mrp_subcontracting_purchase/models/purchase_order.py
@@ -5,7 +5,7 @@ from odoo import models, fields, api
 
 
 class PurchaseOrder(models.Model):
-    _inherit = ['purchase.order']
+    _inherit = 'purchase.order'
 
     subcontracting_resupply_picking_count = fields.Integer(
         "Count of Subcontracting Resupply", compute='_compute_subcontracting_resupply_picking_count',

--- a/addons/mrp_subcontracting_purchase/models/stock_move.py
+++ b/addons/mrp_subcontracting_purchase/models/stock_move.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class StockMove(models.Model):
-    _inherit = ['stock.move']
+    _inherit = 'stock.move'
 
     def _is_purchase_return(self):
         res = super()._is_purchase_return()

--- a/addons/mrp_subcontracting_purchase/models/stock_picking.py
+++ b/addons/mrp_subcontracting_purchase/models/stock_picking.py
@@ -5,7 +5,7 @@ from odoo import models, fields, api, _
 
 
 class StockPicking(models.Model):
-    _inherit = ['stock.picking']
+    _inherit = 'stock.picking'
 
     subcontracting_source_purchase_count = fields.Integer(
         "Number of subcontracting PO Source", compute='_compute_subcontracting_source_purchase_count',

--- a/addons/mrp_subcontracting_purchase/models/stock_rule.py
+++ b/addons/mrp_subcontracting_purchase/models/stock_rule.py
@@ -2,7 +2,7 @@ from odoo import models, _
 
 
 class StockRule(models.Model):
-    _inherit = ['stock.rule']
+    _inherit = 'stock.rule'
 
     def _get_lead_days(self, product, **values):
         """For subcontracting, we need to consider both vendor lead time and

--- a/addons/mrp_subcontracting_purchase/models/stock_valuation_layer.py
+++ b/addons/mrp_subcontracting_purchase/models/stock_valuation_layer.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class StockValuationLayer(models.Model):
-    _inherit = ['stock.valuation.layer']
+    _inherit = 'stock.valuation.layer'
 
     def _get_layer_price_unit(self):
         """ For a subcontracted product, we want a way to get the subcontracting cost (the price on the PO)
